docs/question.md

```diff
@@ -438,12 +443,46 @@ Example of valid HTML:
+
+### Custom Question Best Practices
+
+- Although custom questions usually don't use the grading functions from individual elements, it is _highly_ recommended that built-in elements are used for student input. Parsed student answers are present in the `data["submitted_answers"]` dictionary.
+
+- Any custom grading function for the whole question should set `data["score"]` as a value between 0.0 and 1.0, which will be the final score for the given question.
+
+- If a custom grading function is only grading a specific part of a question, the grading function should set the corresponding dictionary entry in `data["partial_scores"]` and then recompute the final `data["score"]` value for the whole question. To avoid showing score badges for individual parts, the dictionary entries in `data["partial_scores"]` should be unset.
+
+- To set custom feedback, the grading function should set the corresponding entry in the `data["feedback"]` dictionary. These feedback entries are passed in when rendering the `question.html`, which can be accessed by using the mustache prefix `{{feedback.}}`.
+
+- For generated floating point answers, it's important to round values before displaying them to the student, as this could cause unexpected behavior when students perform calculations. For example, the following is problematic:
+
+  ```python
+  def generate(data):
+      a = 33.33337
+      b = 33.33333
+      data["params"]["a_for_student"] = f'{a:.2f}'
+      data["params"]["b_for_student"] = f'{a:.2f}'
+      data["correct_answers"]["c"] = a - b
+  ```
+
+  The rounding should be done before the format strings are set, as in the following example:
+
+  ```python
+  def generate(data):
+    a = np.round(33.33337, 2)
+    b = np.round(33.33333, 2)
+    data["params"]["a_for_student"] = f'{a:.2f}'
+    data["params"]["b_for_student"] = f'{b:.2f}'
+    data["correct_answers"]["c"] = a - b
+  ```
+
+- Similarly, for grading functions involving floating point numbers, _avoid exact comparisons with `==`._ Floating point calculations in Python introduce error, and comparisons with `==` might unexpectedly fail. Instead, the function [`math.isclose`](https://docs.python.org/3/library/math.html#math.isclose) should be used, as it performs comparisons with tolerance values.
```

Comment:
You can also point them to `pl.is_correct_scalar_ra`, `pl.is_correct_scalar_sf` and `pl.is_correct_scalar_dd` for more fine-tuned comparison options.